modules/core/src/main/java/org/apache/ignite/internal/visor/tracing/configuration/VisorTracingConfigurationItem.java

```diff
@@ -0,0 +1,176 @@
+     *  0 and 1 have special meaning here, 0 means never 1 means always. Default value is 0 (never).
+     * @param supportedScopes Set of {@link Scope} that defines which sub-traces will be included in given trace.
+     *  In other words, if child's span scope is equals to parent's scope
+     *  or it belongs to the parent's span supported scopes, then given child span will be attached to the current trace,
+     *  otherwise it'll be skipped.
+     *  See {@link Span#isChainable(org.apache.ignite.internal.processors.tracing.Scope)} for more details.
+     */
+    public VisorTracingConfigurationItem(
+        Scope scope,
+        String lb,
+        Double samplingRate,
+        Set<Scope> supportedScopes)
+    {
+        this.scope = scope;
+        this.lb = lb;
+        this.samplingRate = samplingRate;
+        this.supportedScopes = supportedScopes;
+    }
+
+    /**
+     * @return Specifies the  of a trace's root span to which some specific tracing configuration will be applied. It's
+     * a mandatory attribute.
+     */
+    public Scope scope() {
+        return scope;
+    }
+
+    /**
+     * @return Specifies the label of a traced operation. It's an optional attribute.
+     */
+    public String label() {
+        return lb;
+    }
+
+    /**
+     * @return Number between 0 and 1 that more or less reflects the probability of sampling specific trace. 0 and 1
+     * have special meaning here, 0 means never 1 means always. Default value is 0 (never).
+     */
+    public Double samplingRate() {
+        return samplingRate;
+    }
+
+    /**
+     * @return Set of  that defines which sub-traces will be included in given trace. In other words, if child's span
+     * scope is equals to parent's scope or it belongs to the parent's span supported scopes, then given child span will
+     * be attached to the current trace, otherwise it'll be skipped. See  for more details.
+     */
+    public Set<Scope> supportedScopes() {
+        return supportedScopes;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void writeExternalData(ObjectOutput out) throws IOException {
+        out.writeBoolean(scope != null);
+
+        if (scope != null)
+            out.writeShort(scope.idx());
+
+        U.writeString(out, label());
+
+        out.writeObject(samplingRate);
+
+        U.writeCollection(out, supportedScopes);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void readExternalData(byte protoVer,
+        ObjectInput in) throws IOException, ClassNotFoundException {
+
+        if (in.readBoolean())
+            scope = Scope.fromIndex(in.readShort());
+
+        lb = U.readString(in);
+
+        samplingRate = (Double)in.readObject();
+
+        supportedScopes = U.readSet(in);
+    }
+
+    /** {@inheritDoc} */
+    @Override public String toString() {
+        return S.toString(VisorTracingConfigurationItem.class, this);
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean equals(Object o) {
```

Comment:
It looks like, this method is unnecessary. Am I missing something?

Reply:
Yep, it's used in tests.